src/SetRandomHostPlugin.php

```diff
@@ -19,30 +19,40 @@
 final class SetRandomHostPlugin implements Plugin
 {
     private UriFactoryInterface $uriFactory;
-    /** @var list<string> */
+    /**
+     * A list of hostnames / IP addresses.
+     *
+     * Only protocol, host and port are used. Paths should not be set and are ignored.
+     *
+     * @var list<string>
+     */
     private array $hosts;
     private int $currentHostIndex;
 
     /**
-     * @param array{hosts: string} $config
+     * @param array{hosts: non-empty-array<string>} $config
      */
     public function __construct(UriFactoryInterface $uriFactory, array $config)
     {
         $this->uriFactory = $uriFactory;
         $resolver = new OptionsResolver();
         $resolver->setRequired('hosts');
         $resolver->setAllowedTypes('hosts', 'string[]');
+        $resolver->setAllowedValues('hosts', fn (array $hosts) => (bool) $hosts);
 
         $this->hosts = array_values($resolver->resolve($config)['hosts']);
         $this->currentHostIndex = array_rand($this->hosts);
     }
 
     public function handleRequest(RequestInterface $request, callable $next, callable $first): Promise
     {
+        // would it make sense to cache this? either build all uri in the constructor, or just the one we currently use (in the happy case we will keep using the same)
+        // doing in the constructor would have the added benefit that we validate that the host names can be parsed.
         $uri = $this->uriFactory->createUri($this->hosts[$this->currentHostIndex]);
 
         $request = $request->withUri($request->getUri()
             ->withHost($uri->getHost())
+            // what happens if the uri does not specify a scheme or port?
```

Comment:
I don't think there is much we can do about that. We don't know if value we get from uriFactory is default of that factory or is coming from configured hosts. I'll use some URLs in test case though to make sure it works without, though

Reply:
Ah uriFactory returns empty host in those cases. Addressed.